### PR TITLE
116 mock hubs

### LINKED ISSUE #110: Replace /dhus with /apihub for SciHub
`/apihub` is what we are [supposed to use](https://scihub.copernicus.eu/twiki/do/view/SciHubWebPortal/APIHubDescription) and also the default in `sentinelsat`.

### LINKED ISSUE #116: Use mock requests for testing
@MWieland maybe Github's IP range is blocked (rightly) for the new endpoint and that's why we are getting these errors: `sentinelsat.sentinel.SentinelAPIError: HTTP status 200 OK: API response not valid. JSON decoding failed.`

We would save a lot of troubles by just creating a mock (see https://requests-mock.readthedocs.io/en/latest/overview.html).

_Originally posted by @fwfichtner in https://github.com/dlr-eoc/ukis-pysat/issues/114#issuecomment-801115550_

### PR DESCRIPTION
**Pull request by fwfichtner**
Closes #116 and closes #110

### CODE REVIEW

**Comment by fwfichtner** [comment]
We should consider our own base images for testing, I just don't understand what's wrong

File: tests/test_data.py [hunk 1]
```diff
@@ -4,213 +4,155 @@
...
-    def tearDown(self):
-        catalog_path.unlink()
-
-    def test_init(self):
+    def test_init_stac_catalog(self):
         with Source(datahub=Datahub.STAC_local, catalog=catalog_path) as src:
             self.assertTrue(isinstance(src.api, pystac.catalog.Catalog))
 
+    def test_init_stac_url(self):
+        with Source(datahub=Datahub.STAC_API, url=r"https://earth-search.aws.element84.com/v0/") as src:
```
Comment (MWieland) [inline tests/test_data.py:30]: Aren't we again dependent on the availability of an external datahub here? Perhaps there is a way to also mock this?

File: tests/test_data.py [hunk 2]
```diff
@@ -4,213 +4,155 @@
...
+        )
+
+        with Source(datahub=Datahub.Scihub) as src:
```
Comment (MWieland) [inline tests/test_data.py:68]: I have no experience with mocking request, but where do you use variable m and aren't you still querying the external hub with query metadata? Please ignore if you think this is correct.

File: tests/test_data.py [hunk 3]
```diff
@@ -4,213 +4,155 @@
...
+        )
+        m.get(
+            "https://earthexplorer.usgs.gov/inventory/json/v/1.4.1/logout?jsonRequest=%7B%22apiKey%22%3A+%22mockKey%22%7D",
+            text='{"errorCode":null,"error":"","data":true,"api_version":"1.4.1","access_level":"user","catalog_id":"EE","executionTime":0.4}',
+        )
+
+        with Source(datahub=Datahub.EarthExplorer) as src:
```
Comment (MWieland) [inline tests/test_data.py:95]: Same as previous comment

File: tests/test_raster.py [hunk 4]
```diff
@@ -219,11 +219,12 @@ def test_mask_image(self):
...
 
+    @unittest.skip("Skip until we find a better test or this also runs with Github Actions")
     def test_warp(self):
-        self.assertEqual(self.img.dataset.crs, "EPSG:4326")
+        self.assertEqual(self.img.dataset.crs.data, {"init": "epsg:4326"})
 
         self.img.warp("EPSG:3857")
-        self.assertEqual(self.img.dataset.crs, "EPSG:3857")
+        self.assertEqual(self.img.dataset.crs.data, {"init": "epsg:3857"})
         self.assertEqual(self.img.dataset.meta["crs"], "EPSG:3857")
```
Comment (MWieland) [inline tests/test_raster.py:228]: Where is the problem here (why we skip it)? lower case consistency with all instances of EPSG.

**Review by MWieland — APPROVED** [review]
Well done! Looks really good to me. Just a few questions that are likely related to me having no experience with mocking requests. Feel free to ignore if you think it is alright.

File: tests/test_data.py (hunk 1 above)
Comment (fwfichtner) [reply tests/test_data.py:30]: Sure, we could do this. This was just way more reliable so far.

File: tests/test_data.py (hunk 2 above)
Comment (fwfichtner) [reply tests/test_data.py:68]: See https://requests-mock.readthedocs.io/en/latest/mocker.html#decorator

File: tests/test_raster.py (hunk 4 above)
Comment (fwfichtner) [reply tests/test_raster.py:224]: Yes, this is all a nightmare and the tests suddenly don't work anymore. Best option might be to find the base image and check what is happening there locally.

In this case the return value is lowercase though (should it work...).

File: tests/test_data.py (hunk 1 above)
Comment (fwfichtner) [reply tests/test_data.py:30]: Also we are not using another library for the actual connection here, so I would suggest we keep it until it makes problems.

**Review by MWieland — APPROVED** [review]
Looks good to me. Also cleaner solutions with new usgs api.

**Pull request merged**